File: task/twopointers/ContainerWithMostWater.java
package twopointers;

public class ContainerWithMostWater {
    public static int maxArea(int[] height) {
        int left = 0,
                right = height.length - 1,
                max = 0;

        while (left != right) {
            max = Math.max(max, Math.min(height[left], height[right]) * (right - left));
            if (height[left] < height[right]) left++;
            else right--;
        }

        return max;
    }

    public static void main(String[] args) {
        System.out.println(maxArea(new int[]{10, 9, 8, 7, 6, 5, 4, 3, 2, 1}));
        System.out.println(maxArea(new int[]{1, 8, 6, 2, 5, 4, 8, 3, 7}));
        System.out.println(maxArea(new int[]{1, 1}));
        System.out.println(maxArea(new int[]{4, 3, 2, 1, 4}));
        System.out.println(maxArea(new int[]{1, 2, 1}));
    }
}

File: python/DataEngineeringInterview/ContainerWithMostWater.py
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_area = 0

        left, right = 0, len(height) - 1
        while left < right:
            max_area = max(max_area, height[left] + height[right])
            if height[right] > height[left]:
                left += 1
            else:
                right -= 1

        return max_area

if __name__ == '__main__':
    Solution().maxArea([1,8,6,2,5,4,8,3,7])